api/external/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1/nutanixmachine_types.go

```diff
@@ -93,22 +148,100 @@ type NutanixMachineSpec struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Enum:=legacy;uefi
 	BootType NutanixBootType `json:"bootType,omitempty"`
-
 	// systemDiskSize is size (in Quantity format) of the system disk of the VM
 	// The minimum systemDiskSize is 20Gi bytes
 	// +kubebuilder:validation:Required
 	SystemDiskSize resource.Quantity `json:"systemDiskSize"`
 
+	// dataDisks hold the list of data disks to be attached to the VM
+	// +kubebuilder:validation:Optional
+	DataDisks []NutanixMachineVMDisk `json:"dataDisks,omitempty"`
+
 	// BootstrapRef is a reference to a bootstrap provider-specific resource
 	// that holds configuration details.
 	// +optional
 	BootstrapRef *corev1.ObjectReference `json:"bootstrapRef,omitempty"`
-
 	// List of GPU devices that need to be added to the machines.
 	// +kubebuilder:validation:Optional
 	GPUs []NutanixGPU `json:"gpus,omitempty"`
 }
 
+// NutanixMachineVMDisk defines the disk configuration for a NutanixMachine
+type NutanixMachineVMDisk struct {
```

Comment:
can we have capx bump in separate PR and stack this changes on top of it.

Reply:
These are automated changes that are pulled in so we can't really.